src/builtins/core/timezone.rs

```diff
@@ -105,7 +105,8 @@ impl TimeZone {
     }
 
     pub fn try_from_str(src: &str) -> TemporalResult<Self> {
-        if let Ok(timezone) = Self::try_from_identifier_str(src) {
+        let normalized = src.to_uppercase();
```

Comment:
Oh, wait this is probably what's causing the failure. 

So this is actually sort of a tricky case.

If you upper case a time zone string, "America/New_York" becomes "AMERICA/NEW_YORK". The file system will throw on this because it's case sensitive (although, this may actually depend on your OS configuration). That's actually what #251 is working towards, providing a case insensitive way to return the normalized time zone strings.